ignite/handlers/checkpoint.py

```diff
@@ -692,10 +692,10 @@ def _save_func(self, checkpoint: Mapping, path: str, func: Callable, rank: int =
         else:
             tmp_file = None
             tmp_name = ""
-            tmp = None  # type: _TemporaryFileWrapper
+            tmp = None 
```

Comment:
This
```suggestion
            tmp : Optional[IO[bytes]] = None
```
with 

```python
                tmp_file = tmp.file # type: ignore
```

Should work

Reply:
@trsvchn but IO is not defined anywhere

Reply:
Yes, import it from typing, like we did for Optional etc

Reply:
Yes got it. Thanks a lot @trsvchn